src/System.Management.Automation/engine/CommandSearcher.cs

```diff
@@ -1578,5 +1595,9 @@ internal enum SearchResolutionOptions
         ResolveFunctionPatterns = 0x02,
         CommandNameIsPattern = 0x04,
         SearchAllScopes = 0x08,
+        /// <summary>
+        /// Enable searching for cmdlets/functions by abbreviation expansion.
+        /// </summary>
+        UseAbbreviationExpansion = 0x10,
```

Comment:
`CommandNameIsPattern` set only if the [command name contains wildcards](https://github.com/PowerShell/PowerShell/blob/d80154430d600a52a1d1e3a9f3ecf032b36185d4/src/System.Management.Automation/engine/GetCommandCommand.cs#L720).  In this case when using abbreviation expansion, the command name sans wild card still acts like a pattern.

Reply:
But you changed [that check](https://github.com/PowerShell/PowerShell/blob/d80154430d600a52a1d1e3a9f3ecf032b36185d4/src/System.Management.Automation/engine/GetCommandCommand.cs#L717) to `bool isPattern = WildcardPattern.ContainsWildcardCharacters(plainCommandName) || UseAbbreviationExpansion;`, which means `UseAbbreviationExpansion` implies `CommandNameIsPattern`.

Reply:
@daxian-dbw It's not that `UseAbbreviationExpansion` implies `ResolveFunctionPatterns`.  The default if `-CommandType` is not specified is [`All`](https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/GetCommandCommand.cs#L198) which [includes](https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/GetCommandCommand.cs#L711) `ResolveFunctionPatterns`.  `CommandNameIsPattern` is part of the same enum (incorrectly in my opinion) and used for a different purpose.  It's to indicate that the commandName is a filter rather than literal.  Since `UseAbbreviationExpansion` is also being used as a filter, I added the line to treat it as `isPattern` so that it goes through the filtering (wildcard matching) code path.